src/main/java/org/opensearch/knn/index/query/ResultUtil.java

```diff
@@ -58,19 +57,18 @@ public static void reduceToTopK(List<Map<Integer, Float>> perLeafResults, int k)
     }
 
     /**
-     * Convert map to bit set, if resultMap is empty or null then returns an Optional. Returning an optional here to
-     * ensure that the caller is aware that BitSet may not be present
+     * Convert map of docs to doc id set iterator
      *
      * @param resultMap Map of results
-     * @return BitSet of results; null is returned if the result map is empty
+     * @return Doc id set iterator
      * @throws IOException If an error occurs during the search.
      */
-    public static BitSet resultMapToMatchBitSet(Map<Integer, Float> resultMap) throws IOException {
-        if (resultMap == null || resultMap.isEmpty()) {
-            return null;
+    public static DocIdSetIterator resultMapToDocIds(Map<Integer, Float> resultMap) throws IOException {
+        if (resultMap.isEmpty()) {
+            return DocIdSetIterator.empty();
```

Comment:
I think we would like to keep this as null in return to ensure that caller can exit out of the code early.

Reply:
We can exit by checking if the cost is zero or not. How about it instead of null?

Reply:
If we return `null`, it could lead to a risk of NPE wherever we use the iterator. There's another option to avoid the overhead of `cost()` call in this cases. I'll switch to using a static instance and remove the use of `cost()`. Let me know if you still prefer returning `null` after my change.

Reply:
Hm. Cannot reuse same iterator. Let me switch to null.

Reply:
> Cannot reuse same iterator. I think it is better for caller to check if result map is empty or not before calling this method instead 

Then whoever is passing the `resultMap` then needs to know that empty check on the map should be done outside. I still think we should go with `null` returns and ensure things are added in java docs. I won't suggest validating `resultMap` as empty in the caller as the `resultMapToDocIds` is a public function, so it needs to handle the resultMap being empty case.

Reply:
Looking at Lucene, DocIdSetIterator.empty() is used pretty frequently, so I think its okay to use that.

Reply:
Would prefer empty over null if there is no performance impact. Its more defensive then passing null around. 

Considering  https://github.com/opensearch-project/k-NN/pull/2283#discussion_r1859878598, there doesn't seem to be a perf impact, I would recommend empty

Reply:
> Would prefer empty over null if there is no performance impact. Its more defensive then passing null around.

There will be performance impact because in non empty case `.cost()` function don't give you `O(1)` runtime.

Reply:
I see, performance takes over in that case. if this is forcing us to have cost() and there is a clear impact for worst case scenarios, empty should be rethought

Reply:
As of now, there is no place where we have to call `.cost()` because we are using empty instead of null.